### DISCUSSION TITLE
Core SVG Icons and Fill Color

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#6902] [Ideas]
Is it possible to add `fill: currentcolor;` to core svg icons to follow the color of their surroundings?
Most use cases of SVG icons are in buttons! It is nice to set the color of text and scg follows the same color.

So I would propose to add `fill: currentcolor;` to core svg icons.

**Comment by pmario** [comment]
If you add `fill: something` to the code of the SVGs, it will be impossible to style them with CSS .. So the only way will be CSS classes. Every core SVG has 2 classes eg: 

https://tiddlywiki.com/#%24%3A%2Fcore%2Fimages%2Fadd-comment
has
`tc-image-add-comment tc-image-button`   where the left class is unique to the image. 

Once the PR #6666 will be merged, there should be more compact possibilities to "call" SVGs with transclusions using custom parameters. ...

**Reply by kookma** [reply]
> If you add `fill: something` to the code of the SVGs, it will be impossible to style them with CSS

Can't we apply colors using

```
<span style-"color:red">{{$:/core/images/some-icon}}</span>
```



> `tc-image-add-comment tc-image-button` where the left class is unique to the image.

Yes, this is very useful to customize the look and feel of an svg icon.

**Comment by pmario** [comment]
As I wrote it will probably be possible with transclusion parameters. @Jermolene ... Am I right here?

**Comment by Alzacon** [comment]
If we add the following rule (with low specificity), we don't have problems to style the SVGs (at least I can see them)
```
svg{
fill: currentcolor;
}
```
- With other selectors we can have a higher specificity to style the SVG.
- If the SVG has inline styles, they can be applied.

I think this can be used to help simplify/reduce the stylesheet in the future.
